### PR TITLE
Feature/milestone 6 - Improve friendship model efficiency

### PR DESCRIPTION
**Pull request by michael-mcgee**
**I'm working on this project alone as requested by student success team**

This pull request includes the following changes:

- Updated models to use 2 records for each friendship
- Updated controllers to user new model methods
- Update model tests for new friendship system
- Updated views to use new model and controller methods
- Cleaned up some un-used config files

Please note that some CSS styling is incomplete.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -39,44 +39,50 @@ def likes?(likeable)
...
-  def confirmed_friendships
-    friendships.map { |f| f.friend if f.confirmed }.compact +
-      inverse_friendships.map { |f| f.user if f.confirmed }.compact
+  def received_requests_from
+    User.joins(:friendships).where(friendships: { confirmed: false, friend_id: id })
   end
 
-  def unconfirmed_friendships
-    sent_friend_requests + received_friend_requests
+  def received_requests_count
```
Comment (mosaaleb) [inline app/models/user.rb]: - [ ] **[OPTIONAL]** [`where`](https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-where) clause can accept array
So we can refactor this to be :
```ruby
User.all_except([self, friends, sent_requests_to, received_requests_from])
```

File: app/models/user.rb [hunk 2]
```diff
@@ -39,44 +39,50 @@ def likes?(likeable)
     likeable.likes.where(user_id: id).any?
   end
 
-  def sent_friend_requests
-    friendships.map { |f| f.friend unless f.confirmed }.compact
+  def friends
```
Comment (mosaaleb) [inline app/models/user.rb]: - [ ] **[OPTIONAL]** We can also use `has many through` for friends association:

```ruby
has_many :confirmed_friends, -> { where(friendships: { confirmed: true }) }, through: :friendships, source: :friend
```

Or use nested joins if you prefer joins:
```ruby
User.joins(friendships: :friend).where(friendships: { confirmed: true, friend_id: id })
```

at the end `has many through` are actually nested joins.
[nested joins](https://guides.rubyonrails.org/active_record_querying.html#joining-nested-associations-single-level)

File: app/models/user.rb (hunk 1 above)
Comment (michael-mcgee) [reply app/models/user.rb]: Thank you for the suggestions, updated accordingly.

File: app/models/user.rb (hunk 2 above)
Comment (michael-mcgee) [reply app/models/user.rb]: Thank you for the suggestions, updated accordingly.

**Pull request merged**